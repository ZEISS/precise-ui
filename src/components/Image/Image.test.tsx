import * as React from 'react';
import { shallow } from 'enzyme';
import { Image } from './';

const image =
  'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==';

describe('<Image />', () => {
  it('should render <Image> without wrapper', () => {
    const wrapper = shallow(<Image src={image} alt="" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <Image> within wrapper component when preloaded', () => {
    const wrapper = shallow(<Image preload src={image} alt="some image" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render default error component when image fails onload', () => {
    const wrapper = shallow(<Image preload src="http://broken-link" alt="some image" />);
    wrapper.setState({ status: 'error' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should use alternative preloader when one provided', () => {
    const preloader = <div>Loading...</div>;
    const wrapper = shallow(<Image preload preloader={preloader} src={image} alt="some image" />);
    wrapper.setState({ status: 'loading' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should use alternative error component when one provided', () => {
    const errorComponent = <div>Ooops something went wrong...</div>;
    const wrapper = shallow(<Image preload error={errorComponent} src={image} alt="some image" />);
    wrapper.setState({ status: 'error' });
    expect(wrapper).toMatchSnapshot();
  });
});
