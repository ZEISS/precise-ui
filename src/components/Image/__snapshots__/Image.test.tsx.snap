// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render <Image> within wrapper component when preloaded 1`] = `
<styled.div>
  <Spinner />
</styled.div>
`;

exports[`<Image /> should render <Image> without wrapper 1`] = `
<styled.img
  alt=""
  src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg=="
/>
`;

exports[`<Image /> should render default error component when image fails onload 1`] = `
<styled.div>
  <styled.span>
    Error loading image from 
    http://broken-link
    .
  </styled.span>
</styled.div>
`;

exports[`<Image /> should use alternative error component when one provided 1`] = `
<styled.div>
  <div>
    Ooops something went wrong...
  </div>
</styled.div>
`;

exports[`<Image /> should use alternative preloader when one provided 1`] = `
<styled.div>
  <div>
    Loading...
  </div>
</styled.div>
`;
